src/routes/todo.routes/deleteToDo.ts

```diff
@@ -1,30 +1,30 @@
 import { Router, Request, Response, NextFunction } from "express";
-import { deleteTodo } from "../../repository/todo.repository/todoRepository";
+
 import { Api404Error } from "../../services/error.services/api404Error";
-import { Api500Error } from "../../services/error.services/api500Error";
 import { httpStatusCodes } from "../../services/error.services/httpStatusCodes";
 import getToDo from "../../services/todo.services/todoServices";
 import { ToDo } from "../../repository/todo.repository/types";
+import { todo } from "../../database/models/todos";
 
 export function deleteToDoRouter(): Router {
   const todoRouter = Router();
   return todoRouter.delete(
     "/:id",
-    (req: Request, res: Response, next: NextFunction) => {
+    async (req: Request, res: Response, next: NextFunction) => {
       try {
-        let toDo = {};
-        getToDo(req.params.id).match(
+        let toDo = {} as ToDo;
+
+        (await getToDo(req.params.id)).match(
           (value) => {
             toDo = value;
-            deleteTodo(toDo as ToDo);
+            todo
```

Comment:
same as above, this logic should be handled in different layers of the application